cpp/src/arrow/record_batch_test.cc

```diff
@@ -1033,15 +1039,44 @@ Result<std::shared_ptr<Array>> BuildArray(
   }
   return builder.Finish();
 }
+struct StringBuilderVisitor {
+  template <typename DataType,
+            typename Builder = typename TypeTraits<DataType>::BuilderType>
+  enable_if_t<is_base_binary_type<DataType>::value, Status> Visit(
+      const DataType&, ArrayBuilder* raw_builder,
+      const std::vector<std::string>& values) {
+    Builder* builder = static_cast<Builder*>(raw_builder);
+    ARROW_RETURN_NOT_OK(builder->AppendValues(values));
+    return Status::OK();
+  }
 
-template <typename ArrowType, typename = enable_if_string<ArrowType>>
-Result<std::shared_ptr<Array>> BuildArray(const std::vector<std::string>& values) {
-  using BuilderType = typename TypeTraits<ArrowType>::BuilderType;
-  BuilderType builder;
-  for (const auto& value : values) {
-    ARROW_RETURN_NOT_OK(builder.Append(value));
+  template <typename DataType,
+            typename Builder = typename TypeTraits<DataType>::BuilderType>
+  enable_if_t<is_binary_view_like_type<DataType>::value ||
+                  std::is_same_v<DataType, FixedSizeBinaryType>,
+              Status>
+  Visit(const DataType&, ArrayBuilder* raw_builder,
+        const std::vector<std::string>& values) {
+    Builder* builder = static_cast<Builder*>(raw_builder);
+    // FixedSizeBinary, StringView, BinaryView do not have AppendValues method
```

Comment:
Can we simplify this code by using `Append()` for `Binary`/`String`?
This is a test. So we can use unoptimized methods.